libagbsyscall/Makefile

```diff
@@ -71,17 +73,30 @@ ifneq ($(PLATFORM),gba)
     endif
 
     ifeq ($(CPU_ARCH),i386)
-        # Use the more legible Intel dialect for x86, without underscores
-        CC1FLAGS += -masm=intel -msse2
-        CPPFLAGS += -D CPU_ARCH_X86=1
-    else
-        CPPFLAGS += -D CPU_ARCH_X86=0
+      ifeq ($(PLATFORM), sdl_win32)
+          # Use the more legible Intel dialect for x86, without underscores
+          CC1FLAGS += -masm=intel -msse2
+          CPPFLAGS += -D CPU_ARCH_X86=1
+      else
+        $(error Unknown platforn $(PLATFORM))
+      endif
+    else 
+      ifeq ($(PLATFORM), sdl)
+        CPP := $(CC1) -E
+      endif
+      CPPFLAGS += -D CPU_ARCH_X86=0
     endif
 endif # PLATFORM != gba
 
+ifeq ($(PLATFORM),gba)
 ifeq ($(CPU_ARCH),arm)
     ASFLAGS := -mcpu=arm7tdmi -mthumb-interwork
 endif
+endif
```

Comment:
I think it makes more sense to flip these two ifs?
A potential DS target would use the flags as well, I think.

Reply:
OK!